src/arrayMethodJoin.js

```diff
@@ -6,6 +6,21 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    const sep = separator !== undefined ? separator : ',';
+
+    return this.reduce((acc, curr, index) => {
+      let currElement = curr;
+
+      if (currElement === null || currElement === undefined) {
+        currElement = '';
+      }
+
+      if (index === this.length - 1) {
+        return acc + String(currElement);
+      } else {
```

Comment:
There is no need to use `else` here because you did return inside your `if` block. So you can just return anything directly after `if` block and this return will work only if your `if` condition did'n match.